STYLE_GUIDE.md

```diff
@@ -0,0 +1,196 @@
+# Style Guide for Pull Requests
+
+To ensure consistency throughout the codebase and to aid any of you that are
+interested in contributing, please follow these instructions when structuring
+your code:
+
+# Indentation 
+
+The most general rule is to try and keep everything aligned to a multiple of two
+spaces, so we always wrap after `=` and `in`, like so:
+
+```haskell
+let
+  foo =
+    if quux xs0 then
+      unlines xs0
+    else
+      unlines xs1
+in
+  length foo
+```
+
+Note the placement of the `then` immediately after the `if condition`, this
+ensures the two space indentation is maintained because the positive case is
+then indented on the following line.
+
+This applies to data type definitions as well:
+
+```haskell
+data Foo =
+  Foo {
+      fooFieldA :: Type 
+    , fooFieldB :: Type
+    } deriving Show
+```
+
+```haskell
+data Bar =
+    A
+  | B
+  | C
+  deriving Show
+```
+
+# Long type signatures are displayed on newlines
+
+Avoid:
+
+```haskell
+function :: (MonadTest m, Applicative f, Traversable t) => t a -> Text -> f (a -> b) -> m ()
```

Comment:
This may have been my style leaking into a discussion from slack. I wrap type signatures as a matter of habit as I find them easier to read down than across, and the separation of constraints into a more obvious section is nice too.

Also any changes to the type will be obvious in diffs as line changes, although I consider that of lesser concern.

I'll add some more explanation and change it to something like "when a type signature becomes so long it no longer fits on a single line, wrap it like this" ... or something along those lines.

Reply:
I guess we should define what we consider a line, I'm partial to 72 or 80 because we have a pretty legible style when wrapping occurs I think. I use a bunch of windows in vim, so having thinner code means I can have more of them side by side.

Reply:
>I'm partial to 72 or 80
:+1:

Reply:
On the topic of line length, this is okay?

> A line length limit of approximately 72-80 characters is suggested, but it is
not a strict rule. A line that is 87 characters long because it needs to be 
won't result in your code being rejected. :)